service/telemetry/otelconftelemetry/tracer.go

```diff
@@ -51,7 +54,12 @@ func createTracerProvider(
 	}
 	otel.SetTextMapPropagator(propagator)
 
-	res := newResource(set.Settings, cfg)
+	var attrs []attribute.KeyValue
+	set.Resource.Attributes().Range(func(k string, v pcommon.Value) bool {
+		attrs = append(attrs, attribute.String(k, v.AsString()))
+		return true
+	})
+	res := sdkresource.NewWithAttributes("", attrs...)
```

Comment:
It would be good to reduce repetition and extract this into a new function, separate from `newResource`.